common/transport/amqp/src/amqp.ts

```diff
@@ -454,232 +485,60 @@ export class Amqp {
-   * @param {Function}   putTokenCallback  Called when the put token operation terminates.
-   */
-  putToken(audience: string, token: string, putTokenCallback: GenericAmqpBaseCallback): void {
-    /*Codes_SRS_NODE_COMMON_AMQP_06_016: [The `putToken` method shall throw a ReferenceError if the `audience` argument is falsy.]*/
-    if (!audience) {
-      throw new ReferenceError('audience cannot be \'' + audience + '\'');
-    }
-
-    /*Codes_SRS_NODE_COMMON_AMQP_06_017: [The `putToken` method shall throw a ReferenceError if the `token` argument is falsy.]*/
-    if (!token) {
-      throw new ReferenceError('token cannot be \'' + token + '\'');
-    }
-
-    /*Codes_SRS_NODE_COMMON_AMQP_06_018: [The `putToken` method shall call the `putTokenCallback` callback (if provided) with a `NotConnectedError` object if the amqp client is not connected when the method is called.]*/
-    if (!this._connected) {
-      this._safeCallback(putTokenCallback, new errors.NotConnectedError('Cannot putToken while disconnected.'));
-    } else if (!this._senders[_putTokenSendingEndpoint] || !this._receivers[_putTokenReceivingEndpoint]) {
-      /*Codes_SRS_NODE_COMMON_AMQP_06_022: [ The `putToken` method shall call the `putTokenCallback` callback (if provided) with a `NotConnectedError` object if the `initializeCBS` has NOT been invoked.]*/
-      this._safeCallback(putTokenCallback, new errors.NotConnectedError('Cannot putToken unless initializeCBS invoked.'));
-    } else {
-      /*Codes_SRS_NODE_COMMON_AMQP_06_005: [The `putToken` method shall construct an amqp message that contains the following application properties:
-      'operation': 'put-token'
-      'type': 'servicebus.windows.net:sastoken'
-      'name': <audience>
-
-      and system properties of
-
-      'to': '$cbs'
-      'messageId': <uuid>
-      'reply_to': 'cbs']
-
-      and a body containing <sasToken>. */
-      let amqpMessage = new AmqpMessage();
-      amqpMessage.applicationProperties = {
-        operation: 'put-token',
-        type: 'servicebus.windows.net:sastoken',
-        name: audience
-      };
-      amqpMessage.body = token;
-      amqpMessage.properties = {
-        to: '$cbs',
-        messageId: uuid.v4(),
-        reply_to: 'cbs'
-      };
-
-      let outstandingPutToken: PutTokenOperation = {
-        putTokenCallback: putTokenCallback,
-        expirationTime: Math.round(Date.now() / 1000) + this._putToken.numberOfSecondsToTimeout,
-        correlationId: amqpMessage.properties.messageId
-      };
-
-      this._putToken.outstandingPutTokens.push(outstandingPutToken);
-      //
-      // If this is the first put token then start trying to time it out.
-      //
-      if (this._putToken.outstandingPutTokens.length === 1) {
-        this._putToken.timeoutTimer = setTimeout(this._removeExpiredPutTokens.bind(this), this._putToken.putTokenTimeOutExaminationInterval);
+  private _getRemainingLinks(): AmqpLink[] {
```

Comment:
>_getRemainingLinks [](start = 10, length = 18)

suggest better name.  This does more than just "get".  It also removes the link from the internal collection.  Surely that's significant enough to warrant a different verb.  (but I don't know what that verb would be :))

Reply:
it's used only once I'll inline the function and remove the naming problem.

---
In reply to: [139534573](https://github.com/Azure/azure-iot-sdk-node/pull/83#discussion_r139534573) [](ancestors = 139534573)